src/app/containers/MostRead/utilities/processMostRead.js

```diff
@@ -1,11 +1,48 @@
 import pathOr from 'ramda/src/pathOr';
+import nodeLogger from '#lib/logger.node';
 import { mostReadRecordIsFresh } from '.';
+import { MOST_READ_DATA_INCOMPLETE_WARNING } from '#lib/logger.const';
+
+const logger = nodeLogger(__filename);
+
+const getMostReadItemData = record => {
+  const cpsHeadline = pathOr(
+    null,
+    ['promo', 'headlines', 'shortHeadline'],
+    record,
+  );
+  const optimoHeadline = pathOr(
+    null,
+    [
+      'headlines',
+      'promoHeadline',
+      'blocks',
+      0,
+      'model',
+      'blocks',
+      0,
+      'model',
+      'text',
+    ],
+    record,
+  );
+  const cpsLocator = pathOr(null, ['promo', 'locators', 'assetUri'], record);
```

Comment:
Do we have the `type` for each item, that would be set to `optimo` for Optimo content? We could have a function for each type instead of trying to access all of these paths for all of the content?

Reply:
Yes, there is a `type` attribute for all optimo content. Added `getCpsItemData()` and `getOptimoItemData()` functions to get data for each type.